lib/rpc/broker.toit

```diff
@@ -40,8 +41,12 @@ class RpcBroker implements SystemMessageHandler_:
     procedures_.remove name
 
   on_message type gid pid message/List -> none:
-    assert: type == SYSTEM_RPC_REQUEST_
     id/int := message[0]
+    if type == SYSTEM_RPC_CANCEL_:
+      queue_.cancel id
```

Comment:
What prevents different tasks from accidentally cancelling each others requests? If they have picked the same ID for a request, then if one cancels, then both are cancelled.

Reply:
Yeah, you're right. We need to check the process id too.